src/creatures/monsters/monster.cpp

```diff
@@ -1024,7 +1024,7 @@ void Monster::onThinkDefense(uint32_t interval)
 				}
 			}
 
-			if (summonCount >= summonBlock.max) {
+			if (summonCount >= summonBlock.max || summonCount >= mType->info.maxSummons) {
```

Comment:
Well, isn't that covered in line 1004?

Reply:
Apparently no. If it was, it would work and the correction you made would not be necessary.

Reply:
That's not entirely true I belive.

summonCount has no way to be higher than info.maxSummons, because, well, it is calculated by iterating over summons - and summons can't be higher than info.maxSummons, because it's checked by line 1004.

Reply:
Correction here was to limit summon by it's own maxCount. It was still limited by the info.maxSummons anyway,

Reply:
(actually line 1011)

Reply:
No, maxSummons just doesn't work. Neither here nor on TFS. The reason I don't know, probably has something to do with the switch to revscript.

Reply:
well, ss you can see it's still limtied by the info.maxSummons, which is 4. The problem is it spawns more of them per block than intended 2 - because it was not assigned through lua. Your `||` does nothing. :shrug:

Reply:
> well, as you can see it's still limtied by the info.maxSummons, which is 4. The problem is it spawns more of them per block than intended 2 - because it was not assigned through lua. Your `||` does nothing. 🤷

Revert your commit, add mine changes and see if my "||" does not do anything.

Reply:
Please don't tag me when cherry picking my changes.